### PR TITLE
Python reformattings

### PR DESCRIPTION
**Pull request by CommanderStorm**
These are the changes to the python files, to make #112 more reviewable

### CODE REVIEW

File: data/external/downloader.py [hunk 1]
```diff
@@ -578,63 +598,62 @@ def _get_cache_path(fname):
...
+    with open(path, "w", encoding="utf-8") as file:
+        json.dump(data, file)
 
 
 def _get_tumonline_api_url(base_target):
-    return "https://campus.tum.de/tumonline/{}/NC_{}".format(
-        base_target,
-        str(random.randint(0, 9999)).zfill(4),
-    )
+    # TODO: WTF is this ???
```
Comment (octycs) [inline data/external/downloader.py:657]: I have absolutely no idea, observed it in the API

File: data/compile.py [hunk 2]
```diff
@@ -64,39 +64,21 @@ def main():
...
-    maps.build_roomfinder_maps(data)
+    maps.roomfinder_maps(data)
 
     logging.info("-- 46 Overlay maps")
     maps.add_overlay_maps(data)
 
-    logging.info(f"-- 50 convert {images.IMAGE_BASE} to webp")
-    convert_to_webp(images.IMAGE_BASE)
-    logging.info("-- 51 resize and crop the images for different resolutions and formats")
+    logging.info("-- 50 resize and crop the images for different resolutions and formats")
```
Comment (octycs) [inline data/compile.py:79]:
```suggestion
    logging.info("-- 50 Convert, resize and crop the images for different resolutions and formats")
```
I would add explicitly that this converts the images as well

**Review by octycs — APPROVED** [review]
Thank you for splitting #112. I have looked over all changes and cannot see any issues for the Python part. I added two comments, but else I think this is ready for merge.

**Comment by octycs** [comment]
Just to know, is this intended just for reference? Should I comment on #112 instead?

File: data/external/downloader.py (hunk 1 above)
Comment (CommanderStorm) [reply data/external/downloader.py:657]:
```suggestion
    # I have no idea, what this magic_string is, or why it exists.. Usage is the same as from TUMOnline..
```

**Comment by CommanderStorm** [comment]
I changed the base to main and am merging just the python reformattings ;)

**Pull request merged**